tests/pass_by_ref_test.py

```diff
@@ -179,6 +179,22 @@ def sub_mixed():
         z = ScratchVar(TealType.uint64)
         return mixed_annotations(x, y, z)
 
+    def user_guide_docs_dynamic_scratch_var_demo() -> Expr:
```

Comment:
I like the idea of having user-guide examples tested as well. In the future we might be able to automate this. But for now, it's worthwhile going with this approach, even if it required maintaining parallel copies.

One suggestion: if you create a separate file `tests/user_guide.py` then it will be easier to find the examples. Also, it will be easier to generate/maintain a semantic E2E test for these once that machinary is in place.

Reply:
@tzaffi Good idea to isolate user guide tests.  I attempted to bring in the feedback via https://github.com/algorand/pyteal/pull/226/commits/2bebeb30ba17f8b36f5dd0403a0ba5b48661c52a.